### PR TITLE
API, folder, factoring scrub

### PR DESCRIPTION
**Pull request by sorvell**
Reorganizes src folders into elements, mixins, legacy, and lib. Collapses some mixins which were not considered useful on their own:
* logging folded into legacy-element-mixin
* annotations and event-listeners folded into template-stamp

### CODE REVIEW

File: src/legacy/polymer.dom.html [hunk 1]
```diff
@@ -338,6 +337,15 @@
     return obj.__domApi;
   };
 
+  let p = Element.prototype;
```
Comment (kevinpschaaf) [inline src/legacy/polymer.dom.html]: Move to top of file

File: src/lib/render-status.html [hunk 2]
```diff
@@ -0,0 +1,66 @@
+<!--
```
Comment (kevinpschaaf) [inline src/lib/render-status.html]: /lib --> /utils
/lib/utils.html -->/utils/mixin.html & /utils/import-href.html

File: src/mixins/gesture-event-listeners.html [hunk 3]
```diff
@@ -7,11 +7,10 @@
 Code distributed by Google as part of the polymer project is also
```
Comment (kevinpschaaf) [inline src/mixins/gesture-event-listeners.html]: split to /utils/gestures.html & /mixins/gesture-event-listeners.html

File: src/mixins/template-stamp.html [hunk 4]
```diff
@@ -383,14 +432,91 @@
...
+    return class TemplateStamp extends superClass {
 
-    return class Annotations extends superClass {
+      constructor() {
+        super();
+        this.$ = null;
+        this._templateNodes = null;
+      }
+
+      _stampTemplate(template) {
```
Comment (kevinpschaaf) [inline src/mixins/template-stamp.html]: Note that this method adds _templateNotes and _templateNodes

File: README.md [hunk 5]
```diff
@@ -265,10 +273,7 @@ id is to use `id`.
...
 * <a name="breaking-boolean-attribute-binidng"></a>Binding a default value of `false` via an _attribute binding_ to a boolean property will not override a default `true` property of the target, due to the semantics of boolean attributes.  In general, property binding should always be used when possible, and will avoid such situations.
-* <a name="breaking-lazyRegister"></a>`lazyRegister` option removed and all meta-programming (parsing template, creating accessors on prototype, etc.) is deferred until the first instance of the element is created
+* <a name="breaking-attribute-property-timing"></a>Any attribute values will take priority over property values set prior to upgrade due to V1 `attributeChangedCallback` timing semantics.  In 1.x properties set prior to upgrade overrode attributes.
```
Comment (kevinpschaaf) [inline README.md]: Done

File: src/elements/dom-if.html [hunk 6]
```diff
@@ -96,26 +97,26 @@
...
       //    (or clear) its template's pending host properties before creating
       //    the instance would also avoid the problem.
-      this._renderDebouncer = Polymer.Debouncer.debounce(
-            this._renderDebouncer
+      this.__renderDebouncer = Polymer.Debouncer.debounce(
+            this.__renderDebouncer
           , Polymer.Async.microTask
-          , this._render.bind(this));
-      Polymer.enqueueDebouncer(this._renderDebouncer);
+          , this.__render.bind(this));
```
Comment (kevinpschaaf) [inline src/elements/dom-if.html]: Done

File: src/mixins/element-mixin.html [hunk 7]
```diff
@@ -0,0 +1,514 @@
...
+   * (2) properties effects and observers are created from it at `finalize` time.
+   * @param {HTMLElement} klass
+   * @private
+   */
+  function ownConfigForClass(klass) {
+    if (!klass.hasOwnProperty(
+      goog.reflect.objectProperty('__ownConfig', klass))) {
+      klass.__ownConfig =
+        klass.hasOwnProperty(goog.reflect.objectProperty('config', klass)) ?
+        klass.config : {};
```
Comment (kevinpschaaf) [inline src/mixins/element-mixin.html]: Nope

File: src/mixins/element-mixin.html [hunk 8]
```diff
@@ -0,0 +1,514 @@
...
+   * Configures a `proto` based on a `config` object.
+   * Leverages `PropertyEffects` to create property accessors and effects
+   * supporting, observers, reflecting to attributes, change notification,
+   * computed properties, and read only properties.
+   * @param {HTMLElement} proto
+   * @param {Object} config
+   * @private
+   */
+  function finalizeConfig(proto, config) {
+    let allProps = propertiesForClass(proto.constructor);
```
Comment (kevinpschaaf) [inline src/mixins/element-mixin.html]: Done

File: src/legacy/legacy-element-mixin.html [hunk 9]
```diff
@@ -23,10 +22,12 @@
 
   var utils = Polymer.Utils;
 
+  const hasColor = (window.chrome && !(/edge/i.test(navigator.userAgent))) || (/firefox/i.test(navigator.userAgent));
```
Comment (kevinpschaaf) [inline src/legacy/legacy-element-mixin.html]: Done

File: src/legacy/polymer.dom.html (hunk 1 above)
Comment (kevinpschaaf) [inline src/legacy/polymer.dom.html]: Done

File: src/mixins/element-mixin.html [hunk 10]
```diff
@@ -0,0 +1,514 @@
...
+    let proto = klass.prototype;
+    if (klass.hasOwnProperty(
+      goog.reflect.objectProperty('is', klass)) && klass.is) {
+      Polymer.telemetry.register(proto);
+    }
+    let config = ownConfigForClass(klass);
+    if (config) {
+      finalizeConfig(proto, config);
+    }
+    if (klass.template) {
```
Comment (azakus) [inline src/mixins/element-mixin.html]: Done!

Comment (kevinpschaaf) [inline src/mixins/element-mixin.html]: Done

File: src/mixins/gesture-event-listeners.html (hunk 3 above)
Comment (kevinpschaaf) [inline src/mixins/gesture-event-listeners.html]: Done

**Pull request merged**